desktop/server/server.js

```diff
@@ -25,18 +25,47 @@ function showFailure( app ) {
 	} );
 }
 
-function startServer() {
-	var boot = require( 'boot' );
-	var http = require( 'http' );
-	var server = http.createServer( boot() );
+function startServer( startedCallback ) {
+	var path = require( 'path' ),
+		fork = require( 'child_process' ).fork,
+		appFile = path.resolve( process.cwd(), 'calypso', 'build', 'bundle-desktop.js' ),
+		env = Object.create( process.env ),
+		startedSignal = 'wp-calypso booted',
+		app;
 
-	debug( 'Server created, binding to ' + Config.server_port );
+	env.PORT = Config.server_port;
+	env.NODE_PATH = 'server:client:.';
 
-	server.listen( Config.server_port, Config.server_host );
+	debug( 'Loading server from file: ' + appFile );
+
+	app = fork( appFile, [], {
+		silent: true,
+		cwd: path.resolve( process.cwd(), 'calypso' ), // Force working directory to calypso root
+		env: env
+	} );
+
+	app.stdout.on( 'data', function( data ) {
+		// We need to wait until the server has booted.
+		// Let's use the "booted" message as a signal that it's ready.
+		var str = data.toString();
+		if ( str.indexOf( startedSignal ) === 0 ) {
```

Comment:
Nice! `process.send` is only available for forked processes (e.g. would not be when running running Calypso as-is), so thinking of something like this:

```
diff --git a/index.js b/index.js
index 02ff4d8..dedba3a 100644
--- a/index.js
+++ b/index.js
@@ -23,6 +23,10 @@ console.info( chalk.cyan( '\nGetting bundles ready, hold on...' ) );
 server = http.createServer( app );
 server.listen( port );

+if ( process.env.CALYPSO_IS_FORK ) {
+       process.send( { boot: 'ready' } );
+}
+
 // Enable hot reloader in development
 if ( config( 'env' ) === 'development' ) {
        hotReloader = require( 'bundler/hot-reloader' );
```

Desktop can set the `env` var when forking + watch for the `boot` message.